### PR TITLE
Update Omnibus tooling to support Python checks defined by a `pyproject.toml` file

### PR DESCRIPTION
**Pull request by ofek**
### Motivation

Part of ongoing effort to modernize integration packaging, see https://github.com/DataDog/integrations-core/pull/11233

### Additional Notes

Usage:

- https://github.com/DataDog/datadog-agent/blob/7.33.x/omnibus/config/software/datadog-agent-integrations-py2.rb#L259-L260
- https://github.com/DataDog/datadog-agent/blob/7.33.x/omnibus/config/software/datadog-agent-integrations-py3.rb#L263-L264

### CODE REVIEW

**Comment by mx-psi** [comment]
CI should be fixed by #10696, I am testing that

File: tasks/agent.py [hunk 1]
```diff
@@ -627,23 +629,43 @@ def omnibus_manifest(
...
+            data = toml.loads(f.read())
 
-    prefix = f'Programming Language :: Python :: {python}'
-    for node in ast.walk(tree):
-        if isinstance(node, ast.keyword) and node.arg == "classifiers":
-            classifiers = ast.literal_eval(node.value)
-            print(any(cls.startswith(prefix) for cls in classifiers), end="")
+        project_metadata = data['project']
+        if 'requires-python' not in project_metadata:
             return
```
Comment (mx-psi) [inline tasks/agent.py:647]: For the Omnibus code above to make sense, we should always either print `True` or `False` or raise an exception. Here we are not doing either

File: releasenotes/notes/omnibus-support-pyproject-file-42582cda387d2999.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+enhancements:
+  - |
+    Update Omnibus tooling to support Python checks defined by
+    a ``pyproject.toml`` file.
```
Comment (mx-psi) [inline releasenotes/notes/omnibus-support-pyproject-file-42582cda387d2999.yaml:5]: Is this a user-facing change? If not, I would get rid of the changelog entry

**Review by apigirl — COMMENTED** [review]
no docs review needed

**Review by mx-psi — APPROVED** [review]

**Review by djmitche — APPROVED** [review]

**Review by truthbk — APPROVED** [review]
Looks good! 📦

**Review by yzhan289 — APPROVED** [review]

**Pull request merged**